### PR TITLE
fix(container): emit components as partials

### PR DESCRIPTION
**Pull request by ematipico**
## Changes

The container API was rendering components as full-fledged pages, which was a bug.

This PR ships a breaking change to the rendered default behaviour of the container API. Now, the components are rendered as partials by default, and users can tweak the behaviour by passing `partial: false` as a new option

## Testing

Added new assertions a new test case

## Docs

https://github.com/withastro/docs/pull/9696

### CODE REVIEW

File: .changeset/soft-rabbits-draw.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+"astro": patch
+---
+
+**BREAKING CHANGE to the experimental Container API only**
+
+Until now, the Container API was rendering the components as they were full-fledged Astro pages.
+
+Full-fledged Astro pages contain `<!DOCTYPE html>` but default. This was not intended, and this patch fixes the behavior.
```
Comment (Princesseuh) [inline .changeset/soft-rabbits-draw.md:9]: My take is still that we shouldn't be adding the doctype for people 😠

**Review by Princesseuh — APPROVED** [review]

File: .changeset/soft-rabbits-draw.md (hunk 1 above)
Comment (sarah11918) [inline .changeset/soft-rabbits-draw.md:9]:
```suggestion
Changes the default page rendering behavior of Astro components in containers, and adds a new option `partial: false` to render full Astro pages as before.

Previously, the Container API was rendering all Astro components as if they were full Astro pages containing `<!DOCTYPE html>` by default. This was not intended, and now by default, all components will render as [page partials](https://docs.astro.build/en/basics/astro-pages/#page-partials): only the contents of the components without a page shell.
```
I'm assuming these will work like "page partials" that we've already documented, so we can link there if that's helpful!

File: .changeset/soft-rabbits-draw.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+"astro": patch
+---
+
+**BREAKING CHANGE to the experimental Container API only**
+
+Until now, the Container API was rendering the components as they were full-fledged Astro pages.
+
+Full-fledged Astro pages contain `<!DOCTYPE html>` but default. This was not intended, and this patch fixes the behavior. 
+
+It's still possible to render the component as a full-fledged Astro page by passing a **new option** called `partial: false` to `renderToString` and `renderToResponse`:
```
Comment (sarah11918) [inline .changeset/soft-rabbits-draw.md:11]:
```suggestion
To render the component as a full-fledged Astro page, pass a new option called `partial: false` to `renderToString()` and `renderToResponse()`:
```

File: .changeset/soft-rabbits-draw.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+```js
+import { experimental_AstroContainer as AstroContainer } from 'astro/containerl';
```
Comment (sarah11918) [inline .changeset/soft-rabbits-draw.md:14]:
```suggestion
import { experimental_AstroContainer as AstroContainer } from 'astro/container';
```

**Review by sarah11918 — COMMENTED** [review]
Just some suggestions for your consideration, @ematipico ! 🙌

**Pull request merged**